spark-commons/src/main/scala/za/co/absa/spark/commons/errorhandling/partials/ErrorHandlingCommon.scala

```diff
@@ -54,4 +57,8 @@ trait ErrorHandlingCommon extends ErrorHandling {
     new Column(CaseWhen(branches))
   }
 
+  def errorColumnType(errorColumn: ErrorColumn, dataFrame: DataFrame): DataType = {
+    dataFrame.schema(s"$errorColumn").dataType
```

Comment:
This does not return what we want. This returns a data type of a column named as the string representation of the `ErrorColumn` type, if a column of such name exists in the provided dataframe.